src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -129,37 +125,66 @@
       <RunWithoutTraits Condition="'$(TargetOS)'=='FreeBSD'" Include="nonfreebsdtests"/>
       <RunWithoutTraits Condition="'$(TargetOS)'=='NetBSD'" Include="nonnetbsdtests"/>
     </ItemGroup>
-    
+
+    <PropertyGroup>
+      <RunnerScriptName Condition="'$(TargetOS)'=='Windows_NT'" >RunTests.cmd</RunnerScriptName>
+      <RunnerTemplateName Condition="'$(TargetOS)'=='Windows_NT'" >RunnerTemplate.Windows.txt</RunnerTemplateName>
+      <RunnerScriptName Condition="'$(TargetOS)'!='Windows_NT'" >RunTests.sh</RunnerScriptName>
+      <RunnerTemplateName Condition="'$(TargetOS)'!='Windows_NT'" >RunnerTemplate.Unix.txt</RunnerTemplateName>
+    </PropertyGroup>
+
     <PropertyGroup>
       <XunitTraitOptions Condition="'@(RunWithTraits)'!=''">$(XunitTraitOptions) -trait category=@(RunWithTraits, ' -trait category=') </XunitTraitOptions>
       <XunitTraitOptions Condition="'@(RunWithoutTraits)'!=''">$(XunitTraitOptions) -notrait category=@(RunWithoutTraits, ' -notrait category=') </XunitTraitOptions>
     </PropertyGroup>
 
     <!-- Replace the {XunitTraitOptions} place holder with the actual traits.  We use the place holder
          because code coverage needs to have a bit of the test command line after the traits (it adds ending quotes
-         to one of its options).  Simply appending the traits would break code coverage. -->
+         to one of its options).  Simply appending the traits would break code coverage. 
+         Additionally, replace CoreRun.exe with ./corerun on Non-Windows OSes (this is the only difference in the command)
+         Future refactoring will allow us to construct this correctly initially, but we don't always know the TargetOS 
+         when the properties are set currently.
+         -->
     <PropertyGroup>
+      <TestCommandLine Condition="'$(TargetOS)'!='Windows_NT'" >$(TestCommandLine.Replace('$(TestHostExecutable)', './corerun'))</TestCommandLine>
```

Comment:
Because we didn't have a great way to ensure TargetOS was setup correctly outside of a target so we moved that part here but still kept the windows default outside the target so it could be used by things like VS.